lib/lhm.rb

```diff
@@ -82,16 +83,29 @@ def cleanup_current_run(run, table_name, options = {})
     Lhm::Cleanup::Current.new(run, table_name, connection, options).execute
   end
 
-  def setup(connection)
-    @@connection = Lhm::Connection.new(connection: connection)
+  # Setups DB connection
+  #
+  # @param [ActiveRecord::Base] connection ActiveRecord Connection
+  # @param [Hash] connection_options Optional options (defaults to: empty hash)
+  # @option connection_options [Boolean] :reconnect_with_consistent_host
+  #   Active / Deactivate ProxySQL-aware reconnection procedure (default to: false)
+  def setup(connection, connection_options = {})
+    @@connection = Connection.new(connection: connection, options: connection_options)
   end
 
-  def connection
-    @@connection ||=
-      begin
-        raise 'Please call Lhm.setup' unless defined?(ActiveRecord)
-        ActiveRecord::Base.connection
-      end
+  # Setups DB connection
+  #
+  # @param [Hash] connection_options Optional options (defaults to: empty hash)
+  # @option connection_options [Boolean] :reconnect_with_consistent_host
+  #   Active / Deactivate ProxySQL-aware reconnection procedure (default to: false)
+  def connection(connection_options = nil)
+    if @@connection.nil?
+      raise 'Please call Lhm.setup' unless defined?(ActiveRecord)
+      @@connection = Connection.new(connection: ActiveRecord::Base.connection, options: connection_options || {})
+    else
+      @@connection.options = connection_options unless connection_options.nil?
```

Comment:
Wait, I thought the idea of this method is to just return @@connection if it exist and error if it doesn't? This kind of overwrite feels like it might be fragile.

Reply:
This is also what I thought, but there was an assignment in there. I didn't want to break the "developer facing" interface, so I kept it as is. 

Original code:
```ruby
  def connection
    @@connection ||=
      begin
        raise 'Please call Lhm.setup' unless defined?(ActiveRecord)
        ActiveRecord::Base.connection
      end
  end
```

The decision came from this thread that we had. [[source](https://github.com/Shopify/lhm/pull/112#discussion_r746954532)]